File: src/main/factory/trafficSource/listTrafficSourcesFactory.ts
import { PrismaTrafficSourceRepository } from "../../../infra/repositories/trafficSource";
import { ListTrafficSourcesUseCase } from "../../../app/useCases/trafficSource/listTrafficSourcesUseCase";
import { ListTrafficSourcesController } from "../../../infra/controllers/trafficSource/listTrafficSourcesController";
import { UserGateway } from "../../../infra/gateways/user";

const prismaTrafficSourceRepository = new PrismaTrafficSourceRepository();
const userGateway = new UserGateway();

const listTrafficSourcesUseCase = new ListTrafficSourcesUseCase(
  prismaTrafficSourceRepository,
  userGateway
);

const listTrafficSourcesController = new ListTrafficSourcesController(
  listTrafficSourcesUseCase
);

const listTrafficSources = {
  handle: listTrafficSourcesController.handle.bind(
    listTrafficSourcesController
  ),
};

export { listTrafficSources };

File: internal/main/factory/trafficSource/listTrafficSourcesFactory.go
package trafficSourceFactories

import (
	trafficSourceUseCases "github.com/Lucas-Eduardo-Goncalves/arkyn-store/internal/app/useCases/trafficSource"
	trafficSourceControllers "github.com/Lucas-Eduardo-Goncalves/arkyn-store/internal/infra/controllers/trafficSource"
	trafficSourceRepositories "github.com/Lucas-Eduardo-Goncalves/arkyn-store/internal/infra/repositories"
	contracts "github.com/Lucas-Eduardo-Goncalves/arkyn-store/internal/main/contracts"
)

func NewListTrafficSourcesFactory() contracts.Controller {
	trafficSourceRepository := trafficSourceRepositories.NewInMemoryTrafficSourceRepository()
	listTrafficSourceUseCase := trafficSourceUseCases.NewListTrafficSourcesUseCase(trafficSourceRepository)
	return trafficSourceControllers.NewListTrafficSourcesController(listTrafficSourceUseCase)
}
